weave-js/src/common/components/FileBrowser.tsx

```diff
@@ -608,103 +608,94 @@ interface CodePreviewProps {
-  const [data, setDataVal] = useState('');
-  const [error, setErrorVal] = useState<string | undefined>(undefined);
-  const ref = useRef<HTMLDivElement>(null);
-  const setData = useCallback(
-    (d: string) => {
-      // Automatically reformat JSON
-      let lines = d.split('\n');
-      if (
-        (file.name.endsWith('.json') && lines.length === 1) ||
-        (lines.length === 2 && lines[1] === '')
-      ) {
-        try {
-          const parsed = JSON.parse(lines[0]);
-          lines = JSON.stringify(parsed, undefined, 2).split('\n');
-        } catch {
-          // ok
-        }
-      }
-
-      // Truncate long lines
-      const truncated = lines
-        .map(line => {
-          if (line.length > 1000) {
-            return line.slice(0, 1000) + ' (line truncated to 1000 characters)';
-          } else {
-            return line;
+const CodePreview: FC<CodePreviewProps> = memo(
+  ({useLoadFile, file, language}) => {
+    const [data, setDataVal] = useState('');
+    const [error, setErrorVal] = useState<string | undefined>(undefined);
+    const ref = useRef<HTMLDivElement>(null);
+    const setData = useCallback(
+      (d: string) => {
+        // Automatically reformat JSON
+        let lines = d.split('\n');
+        if (
+          (file.name.endsWith('.json') && lines.length === 1) ||
+          (lines.length === 2 && lines[1] === '')
+        ) {
+          try {
+            const parsed = JSON.parse(lines[0]);
+            lines = JSON.stringify(parsed, undefined, 2).split('\n');
+          } catch {
+            // ok
           }
-        })
-        .join('\n');
+        }
 
-      setDataVal(truncated);
-    },
-    [setDataVal, file.name]
-  );
-  const setError = useCallback(
-    (errorString?: string) => setErrorVal(errorString || 'Error loading file'),
-    [setErrorVal]
-  );
+        // Truncate long lines
+        const truncated = lines
+          .map(line => {
+            if (line.length > 1000) {
+              return (
+                line.slice(0, 1000) + ' (line truncated to 1000 characters)'
+              );
+            } else {
+              return line;
+            }
+          })
+          .join('\n');
+
+        setDataVal(truncated);
+      },
+      [setDataVal, file.name]
+    );
+    const setError = useCallback(
+      (errorString?: string) =>
+        setErrorVal(errorString || 'Error loading file'),
+      [setErrorVal]
+    );
 
-  // We don't pass a fallback to allow dev mode zero byte files to render
-  const loading = useLoadFile(file, {
-    onSuccess: setData,
-    onFailure: setError,
-  });
-  useEffect(() => {
-    if (ref.current != null) {
-      Prism.highlightElement(ref.current);
+    // We don't pass a fallback to allow dev mode zero byte files to render
+    const loading = useLoadFile(file, {
+      onSuccess: setData,
+      onFailure: setError,
+    });
+    useEffect(() => {
+      if (ref.current != null && file.sizeBytes / 1024 < 1024) {
```

Comment:
ignore highlighting for files larger than 1MB. This can probably be bumped up if we dig into the highlighting solution more